subxt/src/client.rs

```diff
@@ -264,21 +294,63 @@ where
                 .system_account_next_index(signer.account_id())
                 .await?
         };
-        let call = self
-            .client
-            .metadata()
-            .pallet(C::PALLET)
-            .and_then(|pallet| pallet.encode_call(&self.call))?;
-
-        let signed = extrinsic::create_signed(
-            &self.client.runtime_version,
-            self.client.genesis_hash,
+
+        // 2. SCALE encode call data to bytes (pallet u8, call u8, call params).
+        let call_data = {
+            let mut bytes = Vec::new();
+            let pallet = self.client.metadata().pallet(C::PALLET)?;
+            bytes.push(pallet.index());
+            bytes.push(pallet.call_index::<C>()?);
+            self.call.encode_to(&mut bytes);
+            Encoded(bytes)
+        };
+
+        // 3. construct our custom additional/extra params.
+        let additional_and_extra_params = X::new(
+            self.client.runtime_version.spec_version,
+            self.client.runtime_version.transaction_version,
             account_nonce,
-            call,
-            signer,
-            additional_params,
-        )
-        .await?;
-        Ok(signed)
+            self.client.genesis_hash,
+            other_params,
+        );
+
+        // 4. construct signature
+        let signature = {
+            let mut bytes = Vec::new();
+            call_data.encode_to(&mut bytes);
+            additional_and_extra_params.encode_extra_to(&mut bytes);
+            additional_and_extra_params.encode_additional_to(&mut bytes);
+            if bytes.len() > 256 {
+                signer.sign(&sp_core::blake2_256(&bytes))
+            } else {
+                signer.sign(&bytes)
+            }
+        };
+
+        // 5. Encode extrinsic, now that we have the parts we need.
+        // (this may not be 100% correct but should be close enough for this example)
+        let extrinsic = {
+            let mut encoded_inner = Vec::new();
+            // "is signed" + transaction protocol version (4)
+            (0b10000000 + 4u8).encode_to(&mut encoded_inner);
+            // from address for signature
+            signer.address().encode_to(&mut encoded_inner);
+            // the signature bytes
+            signature.encode_to(&mut encoded_inner);
+            // attach custom extra params
+            additional_and_extra_params.encode_extra_to(&mut encoded_inner);
+            // and now, call data
+            call_data.encode_to(&mut encoded_inner);
+            // now, prefix byte length:
+            let len = Compact(encoded_inner.len() as u64);
```

Comment:
Needs to be `u32` as per https://github.com/paritytech/substrate/blob/master/primitives/runtime/src/generic/unchecked_extrinsic.rs#L244. So something like:

```suggestion
           let len = Compact(u32::try_from(encoded_inner.len())?);
```

Though you probably will need to `map_err`

Reply:
Being compact encoded it doesn't really matter what the original type is, does it (u8, u16, u32, u64 etc) since they all encode to the same compact bit representation anyway?

Reply:
(but a length that didn't actually fit into a u32 wouldn't decode into a `Compact<u32>` of course)